resources/constructions.rb

```diff
@@ -1380,25 +1319,37 @@ def self.create_kiva_slab_foundation(model, int_horiz_r, int_horiz_width, int_ve
     return foundation
   end
 
-  def self.apply_kiva_walled_foundation(model, int_vert_r, int_vert_depth,
-                                        ext_vert_r, ext_vert_depth,
-                                        wall_height_above_grade)
+  def self.apply_kiva_walled_foundation(model, wall_height, ext_vert_r, int_vert_r,
+                                        ext_vert_offset, int_vert_offset, ext_vert_depth, int_vert_depth,
+                                        wall_height_above_grade, wall_concrete_thick_in, wall_drywall_thick_in)
 
     # Create the Foundation:Kiva object for crawl/basement foundations
     foundation = OpenStudio::Model::FoundationKiva.new(model)
 
     # Interior vertical insulation
-    if int_vert_r > 0 and int_vert_depth > 0
-      int_vert_mat = create_insulation_material(model, "FoundationIntVertIns", int_vert_r)
-      foundation.setInteriorVerticalInsulationMaterial(int_vert_mat)
-      foundation.setInteriorVerticalInsulationDepth(UnitConversions.convert(int_vert_depth, "ft", "m"))
+    if not int_vert_r.nil?
+      if int_vert_r > 0 and int_vert_depth > 0
```

Comment:
Just combine into a single if statement with 3 clauses, no need for nested if statements.